filebeat/harvester/log.go

```diff
@@ -271,6 +271,10 @@ func (h *Harvester) close() {
 	harvesterClosed.Add(1)
 }
 
+// newLogFileReader creates a new reader to read log files
+// It creates a chain of readers which looks as following:
+// limit -> (multiline -> timeout) -> strip_newline -> json -> encode -> log_file
```

Comment:
reverse order to match reading order from lowest-level to highest. This makes actual data flow more clear.

```
logFile -> encoding -> json -> strip_newline -> (timeout -> multiline) -> limit
```